r2golf installation issue

I'm on a debian system with radare2 recently (this week) installed grom github.
```
r2pm install r2golf                           
Already up-to-date.
Install Done For r2golf
cc -g -I/usr/include/libr -lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl -shared -o r2golf.so plugin.c
/usr/bin/ld: /tmp/ccXIF8ko.o: relocation R_X86_64_PC32 against symbol `md_next' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
Makefile:7: recipe for target 'all' failed
make: *** [all] Error 1
mkdir -p "/home/michele/.local/share/radare2/plugins"
cp -f r2golf.so "/home/michele/.local/share/radare2/plugins"
cp: cannot stat 'r2golf.so': No such file or directory
Makefile:10: recipe for target 'install' failed
make: *** [install] Error 1
```

I also give it a try by manually cloning this repo and manually executing golf makefile
```
$ git clone https://github.com/0ki/r2lrn
Cloning into 'r2lrn'...
remote: Enumerating objects: 180, done.
remote: Counting objects: 100% (180/180), done.
remote: Compressing objects: 100% (111/111), done.
remote: Total 180 (delta 60), reused 160 (delta 47), pack-reused 0
Receiving objects: 100% (180/180), 63.44 KiB | 0 bytes/s, done.
Resolving deltas: 100% (60/60), done.
 
$  cd r2lrn/_other/golf-plugin 

$ make
cc -g -I/usr/include/libr -lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl -shared -o r2golf.so plugin.c
/usr/bin/ld: /tmp/ccwM6DrE.o: relocation R_X86_64_PC32 against symbol `md_next' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
Makefile:7: recipe for target 'all' failed
make: *** [all] Error 1
```

Interesting to note as libr include path seems missing
```
$ pkg-config --libs r_core
-lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl

$ pkg-config --cflags r_core
-I/usr/include/libr

$ ls -lai /usr/include/libdr   
ls: cannot access '/usr/include/libdr': No such file or directory
```

Libdr != libr

Which version of r2 do you have? 

> On 22 Sep 2019, at 23:16, Michele <notifications@github.com> wrote:
> 
> ﻿
> I'm on a debian system with radare2 recently (this week) installed grom github.
> 
> r2pm install r2golf                           
> Already up-to-date.
> Install Done For r2golf
> cc -g -I/usr/include/libr -lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl -shared -o r2golf.so plugin.c
> /usr/bin/ld: /tmp/ccXIF8ko.o: relocation R_X86_64_PC32 against symbol `md_next' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: Bad value
> collect2: error: ld returned 1 exit status
> Makefile:7: recipe for target 'all' failed
> make: *** [all] Error 1
> mkdir -p "/home/michele/.local/share/radare2/plugins"
> cp -f r2golf.so "/home/michele/.local/share/radare2/plugins"
> cp: cannot stat 'r2golf.so': No such file or directory
> Makefile:10: recipe for target 'install' failed
> make: *** [install] Error 1
> I also give it a try by manually cloning this repo and manually executing golf makefile
> 
> $ git clone https://github.com/0ki/r2lrn
> Cloning into 'r2lrn'...
> remote: Enumerating objects: 180, done.
> remote: Counting objects: 100% (180/180), done.
> remote: Compressing objects: 100% (111/111), done.
> remote: Total 180 (delta 60), reused 160 (delta 47), pack-reused 0
> Receiving objects: 100% (180/180), 63.44 KiB | 0 bytes/s, done.
> Resolving deltas: 100% (60/60), done.
>  
> $  cd r2lrn/_other/golf-plugin 
> 
> $ make
> cc -g -I/usr/include/libr -lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl -shared -o r2golf.so plugin.c
> /usr/bin/ld: /tmp/ccwM6DrE.o: relocation R_X86_64_PC32 against symbol `md_next' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: Bad value
> collect2: error: ld returned 1 exit status
> Makefile:7: recipe for target 'all' failed
> make: *** [all] Error 1
> Interesting to note as libr include path seems missing
> 
> $ pkg-config --libs r_core
> -lr_core -lr_config -lr_debug -lr_bin -lr_anal -lr_bp -lr_egg -lr_asm -lr_lang -lr_parse -lr_flag -lr_cons -lr_reg -lr_search -lr_syscall -lr_fs -lr_io -lr_socket -lr_magic -lr_crypto -lr_hash -lr_util -ldl
> 
> $ pkg-config --cflags r_core
> -I/usr/include/libr
> 
> $ ls -lai /usr/include/libdr   
> ls: cannot access '/usr/include/libdr': No such file or directory
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

--- a/_other/golf-plugin/Makefile
+++ b/_other/golf-plugin/Makefile
@@ -4,7 +4,7 @@ LDFLAGS=$(shell pkg-config --libs r_core)
 R2_USER_PLUGINS=$(shell r2 -H R2_USER_PLUGINS)
 
 all:
-	$(CC) -g $(CFLAGS) $(LDFLAGS) -shared -o r2golf.$(LIBEXT) plugin.c
+	$(CC) -g $(CFLAGS) $(LDFLAGS) -fPIC -shared -o r2golf.$(LIBEXT) plugin.c
 
 install:
 	mkdir -p "$(R2_USER_PLUGINS)"

